src/main/cpp/src/TaskPriorityJni.cpp

```diff
@@ -18,16 +18,43 @@
 #include "jni_utils.hpp"
 #include "task_priority.hpp"
 
+namespace {
+// Track the next priority to assign and maintain a map of attempt_id to priority
+static long next_task_priority = std::numeric_limits<long>::max() - 1;
+static std::mutex priority_mutex;
+static std::unordered_map<long, long> attempt_priorities;
+}  // namespace
+
 namespace spark_rapids_jni {
 
 long get_task_priority(long attempt_id)
 {
-  return std::numeric_limits<long>::max() - (attempt_id + 1);
+  if (attempt_id == -1) {
+    // Special case: -1 always gets highest priority
+    return std::numeric_limits<long>::max();
```

Comment:
so -1 is meant to be higher than the first task? (which is max() - 1)? I assume this is the intent.

Reply:
yes this matches the code that was there before. -1 is used to indicate a thread with no task associated with it.

Reply:
Presumably if we have work going on in a thread not associated with the task we don't know how to prioritize it, yet. This is a default chosen to not block those tasks/and to not spill that data first. We should experiment with other defaults here.

It may be that we arrive at a place where not thread has -1 priority, but we need to experiment.